raiden/api/rest.py

```diff
@@ -1326,6 +1329,131 @@ def get_pending_transfers(
+
+        return api_response(
+            status_code=HTTPStatus.OK, result=dict(transaction_hash=encode_hex(transaction_hash))
+        )
+
+    def _plan_withdraw_from_udc(self, planned_withdraw_amount: TokenAmount) -> Response:
+        log.debug(
+            "Planning a withdraw from UDC",
+            node=self.checksum_address,
+            planned_withdraw_amount=planned_withdraw_amount,
+        )
+
+        try:
+            (transaction_hash, planned_withdraw_block_number) = self.raiden_api.plan_udc_withdraw(
+                planned_withdraw_amount
+            )
+        except InsufficientEth as e:
+            return api_error(errors=str(e), status_code=HTTPStatus.PAYMENT_REQUIRED)
+        except WithdrawMismatch as e:
+            return api_error(errors=str(e), status_code=HTTPStatus.CONFLICT)
+        except UserDepositNotConfigured as e:
+            return api_error(errors=str(e), status_code=HTTPStatus.NOT_FOUND)
+        except RaidenRecoverableError as e:
+            return api_error(errors=str(e), status_code=HTTPStatus.CONFLICT)
+
+        result = dict(
+            transaction_hash=encode_hex(transaction_hash),
+            planned_withdraw_block_number=planned_withdraw_block_number,
+        )
+        return api_response(status_code=HTTPStatus.OK, result=result)
+
+    def _withdraw_from_udc(self, amount: TokenAmount) -> Response:
+        log.debug(
+            "Withdraw from UDC",
+            node=self.checksum_address,
+            amount=amount,
+        )
+
+        try:
+            transaction_hash = self.raiden_api.withdraw_from_udc(amount)
+        except InsufficientEth as e:
+            return api_error(errors=str(e), status_code=HTTPStatus.PAYMENT_REQUIRED)
+        except WithdrawMismatch as e:
+            return api_error(errors=str(e), status_code=HTTPStatus.CONFLICT)
+        except UserDepositNotConfigured as e:
+            return api_error(errors=str(e), status_code=HTTPStatus.NOT_FOUND)
+        except RaidenRecoverableError as e:
+            return api_error(errors=str(e), status_code=HTTPStatus.CONFLICT)
+
+        return api_response(
+            status_code=HTTPStatus.OK, result=dict(transaction_hash=encode_hex(transaction_hash))
+        )
+
+    def send_udc_transaction(
+        self,
+        total_deposit: TokenAmount = None,
+        planned_withdraw_amount: TokenAmount = None,
+        withdraw_amount: TokenAmount = None,
+    ) -> Response:
+        log.debug(
+            "Sending UDC transaction",
+            node=self.checksum_address,
+            total_deposit=total_deposit,
+            planned_withdraw_amount=planned_withdraw_amount,
+            withdraw_amount=withdraw_amount,
+        )
+
+        if total_deposit is not None and planned_withdraw_amount is not None:
+            return api_error(
+                errors="Cannot deposit to UDC and plan a withdraw at the same time",
+                status_code=HTTPStatus.BAD_REQUEST,
+            )
```

Comment:
I don't like such multi-purpose functions, but since other API endpoints work in the same way, this seems to be the right thing to do for consistency's sake.

Reply:
Yep, I agree. We should probably improve in that sense in a v2 of the API.